### PR TITLE
[NUI] Add Constraint base + inhouse ResizePolicyType:KeepSizeFollowingParent

### PR DESCRIPTION
**Pull request by hinohie**
Add Constraint basic logics +

Add ResizePolicyType
 - KeepSizeFollowingParent

NOTE : This ResizePolicy Breakout layout features when we use both KeepSizeFollowingParent and Layout. Becareful

This patch need to merge below dali patch
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/261156/

### CODE REVIEW

File: src/Tizen.NUI/src/internal/Animation/Constraint.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    using global::System;
+    using global::System.Runtime.InteropServices;
+
+    /// <summary>
+    /// An abstract base class for Constraints.
+    /// This class only use for inhouse currently.
+    /// 
+    /// This can be used to constrain a property of an object, after animations have been applied.
+    /// Constraints are applied in the following order:
+    ///  - Constraints are applied to on-stage actors in a depth-first traversal.
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/internal/Animation/Constraint.cs:30]: there is no actor in NUI

File: src/Tizen.NUI/src/internal/Animation/Constraint.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    ///  - For each actor, the constraints are applied in the same order as the calls to Apply().
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/internal/Animation/Constraint.cs:31]: there is no actor in NUI

**Review by hyunjushin — APPROVED** [review]

File: src/Tizen.NUI/src/internal/Animation/Constraint.cs (hunk 2 above)
Comment (hinohie) [reply src/Tizen.NUI/src/internal/Animation/Constraint.cs]: fix done.

File: src/Tizen.NUI/src/internal/Animation/Constraint.cs (hunk 1 above)
Comment (hinohie) [reply src/Tizen.NUI/src/internal/Animation/Constraint.cs]: fix done.

**Review by Seoyeon2Kim — APPROVED** [review]
I wanna get other NUI members' reviews for an additional `ResizePolicyType`.

**Pull request merged**